### PR TITLE
Update UserGuide.md

### PR DESCRIPTION
**Pull request by heeeyi**
Updated UserGuide.md. Added:
* In-file links
* `Edit by NRIC` commands
* Auto-complete feature (pls help to review)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
+* Only elderly members and volunteers with intersecting available dates can be paired.
+* Elderly member and volunteers in different regions can be paired but a warning message is issued.
+* Duplicate pairs will fail to be added to FriendlyLink.
+* Alphabets in NRIC are case-insensitive.
+
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Notes pairing**<br>
+
+* Pairing will only be successful if there are no clashes in availability that is specified by the elderly and volunteer.
```
Comment (zhacatomn) [inline docs/UserGuide.md:169]: I believe this restriction is being removed. See #140

File: docs/UserGuide.md [hunk 2]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
+* When editing fields allowing multiple inputs, the existing contents of the field will be removed. i.e. editing of tags or available dates will overwrite previous ones and are not cumulative.
+* You can remove all the elderly’s tags by typing `t/` without specifying any tags after it. Same for available dates by typing `dr/`.
 
 Examples:
+
 * `edit_elderly 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st elderly to be `91234567` and `johndoe@example.com` respectively.
 * `edit_elderly 2 n/Betsy Crower t/` Edits the name of the 2nd elderly to be `Betsy Crower` and clears all existing tags.
 
-### Finding people and their related pairs: `find`
+#### Editing a volunteer by index : `edit_volunteer`
```
Comment (zhacatomn) [inline docs/UserGuide.md:200]:
```suggestion
#### Editing a volunteer by index: `edit_volunteer`
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
-* Deletes the volunteer with the specified NRIC `NRIC`.
-
-Examples:
-* `delete_volunteer S8238658J` deletes the volunteer with NRIC `S8238658J`.
+### Deleting records
 
+Delete the specific information of an existing elderly or volunteer in FriendlyLink, based on their NRIC.
 
-### Pair volunteer and elderly: `add_pair`
+#### Deleting an elderly : `delete_elderly`
```
Comment (zhacatomn) [inline docs/UserGuide.md:241]:
```suggestion
#### Deleting an elderly: `delete_elderly`
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
+* If the deleted elderly has existing pairings, the corresponding pairing information will also be deleted.
 
-* After pairing, the newly added pairs appear in the pair list in the window.
-* Only elderly members and volunteers existing in FriendlyLink's data can be paired.
-* Only elderly members and volunteers with intersecting available dates can be paired.
-* Elderly member and volunteers in different regions can be paired but a warning message is issued.
-* Duplicate pairs will fail to be added to FriendlyLink.
-* Alphabets in NRIC are case-insensitive.
+Examples:
+* `delete_elderly S8238657A` deletes an existing elderly with NRIC `S8238657A`.
```
Comment (zhacatomn) [inline docs/UserGuide.md:252]:
```suggestion
* `delete_elderly S8238657A` deletes an existing elderly with NRIC `S8238657A`, as well as all the pairings containing this elderly.
```
Or something to this effect.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
 
-<div markdown="block" class="alert alert-info">
+#### Deleting a volunteer : `delete_volunteer`
```
Comment (zhacatomn) [inline docs/UserGuide.md:254]:
```suggestion
#### Deleting a volunteer: `delete_volunteer`
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -86,174 +92,217 @@ Shows a message explaining how to access the help page.
...
+
+### Finding people and their related pairs: `find`
+
+Finds any elderly or volunteers matching **all** of the specified fields, and pairings that they are involved in.
+
+Format: `find [n/NAME] [nr/NRIC] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [ag/AGE] [re/REGION] [r/RISK_LEVEL] [mt/MEDICAL_QUALIFICATIONS] [t/TAG] [dr/AVAILABLE_DATE_START, AVAILABLE_DATE_END]`
+
+* Fields can be in any order.
+* The fields are optional so any combination of them is possible but **at least one** field must be specified.
+* When multiple of the same field is specified, the last one will be taken. e.g. `find n/jane n/john` will search for names that contain `john` only.
```
Comment (zhacatomn) [inline docs/UserGuide.md:292]:
```suggestion
* When multiple of the same field are specified, only the last one will be taken. e.g. `find n/jane n/john` will search for names that contain `john` only.
```

File: docs/UserGuide.md [hunk 7]
```diff
@@ -315,33 +387,51 @@ If your changes to the data file makes its format invalid, FriendlyLink will dis
...
+| Action               | Format, Examples                                                                                                                                                                                                                                                                                           |
+|----------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Add Elderly**      | `add_elderly n/NAME enr/NRIC [p/PHONE] [e/EMAIL] [a/ADDRESS] [ag/AGE] [re/REGION] [r/RISK] [t/TAG]… [dr/AVAILABLE_DATE_START, AVAILABLE_DATE_END]…` <br> e.g.,`add_elderly n/John p/98765432 e/johnd@example.com a/John street enr/S1234567C ag/68 re/NORTH r/HIGH t/lonely dr/2023-06-03,2023-06-17`      |
```
Comment (zhacatomn) [inline docs/UserGuide.md:422]:
```suggestion
| **Add Elderly**      | `add_elderly n/NAME enr/NRIC [p/PHONE] [e/EMAIL] [a/ADDRESS] [ag/AGE] [re/REGION] [r/RISK_LEVEL] [t/TAG]… [dr/AVAILABLE_DATE_START, AVAILABLE_DATE_END]…` <br> e.g.,`add_elderly n/John p/98765432 e/johnd@example.com a/John street enr/S1234567C ag/68 re/NORTH r/HIGH t/lonely dr/2023-06-03,2023-06-17`      |
```

**Review by zhacatomn — CHANGES_REQUESTED** [review]
Some minor grammar fixes and wording changes, but overall looks good. Thanks for the hard work!

File: docs/UserGuide.md (hunk 1 above)
Comment (heeeyi) [reply docs/UserGuide.md:169]: #140 is not "allowing pairing with no overlapping available dates", but "after edit, one pair may have no overlapping available dates, warn such edits"

Comment (wz2k) [reply docs/UserGuide.md:170]: I feel that add pair should completely block if available dates don't match but also throw a warning like edit. Need @gohyongjing to confirm.

Comment (gohyongjing) [reply docs/UserGuide.md:170]: yeap, my pr #150 makes it such that `editElderly`, `editVolunteer` and `addPair` all accepts pair with no matching dates and regions but gives a notice in the success message

Comment (heeeyi) [reply docs/UserGuide.md:170]: Ok I will then update it

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -119,7 +119,7 @@ How the parsing works:
 ### Model component
 **API** : [`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
 
-<img src="images/ModelClassDiagram.png" width="450" />
+<img src="images/developerGuide/ModelDiagram.png" width="500" />
 ** TODO: update image **
```
Comment (wz2k) [inline docs/DeveloperGuide.md:123]: Can remove

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -129,11 +129,15 @@ The `Model` component,
...
+This is the detailed implementation of `Person` in `Model`. 
+* Both `Elderly` and `Volunteer` inherit from the abstract class `Person`.
+* A `Pair` makes reference to one `Elderly` and one `Volunteer` each.
+
+<br>
+
+<img src="images/developerGuide/PersonNew.png" width="250" />
 
-</div>
 ** TODO: Update Image **
```
Comment (wz2k) [inline docs/DeveloperGuide.md:141]: same here

**Review by wz2k — APPROVED** [review]
Great work 👍

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -123,7 +123,7 @@ How the parsing works:
 ### Model component
 **API** : [`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
 
-<img src="images/ModelClassDiagram.png" width="450" />
+<img src="images/developerGuide/ModelDiagram.png" width="500" />
 ** TODO: update image **
```
Comment (Zxun2) [inline docs/DeveloperGuide.md:127]:
```suggestion

```

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -133,11 +133,15 @@ The `Model` component,
...
+This is the detailed implementation of `Person` in `Model`. 
+* Both `Elderly` and `Volunteer` inherit from the abstract class `Person`.
+* A `Pair` makes reference to one `Elderly` and one `Volunteer` each.
+
+<br>
+
+<img src="images/developerGuide/PersonNew.png" width="250" />
 
-</div>
 ** TODO: Update Image **
```
Comment (Zxun2) [inline docs/DeveloperGuide.md:145]:
```suggestion

```

**Review by Zxun2 — COMMENTED** [review]
Thanks for the good work!

File: docs/UserGuide.md (hunk 1 above)
Comment (Zxun2) [inline docs/UserGuide.md:170]:
```suggestion
* A warning will be given if there are clashes in availability between the volunteer and elderly when pairing.
```

**Comment by Zxun2** [comment]
@heeeyi You can merge it when it's ready 👌🏻

**Pull request merged**